packages/git/src/browser/history/git-history-contribution.ts

```diff
@@ -43,6 +46,27 @@ export class GitHistoryContribution extends AbstractViewContribution<GitHistoryW
         });
     }
 
+    @postConstruct()
+    protected init() {
+        this.repositoryTracker.onDidChangeRepository(async repository => {
```

Comment:
@akosyakov, how should we deal with the disposal of the listeners for such a contribution? Do we have to do anything at all? It is neither a `Widget` nor the contribution itself is a `Disposable`. Thanks!

[Related PR](https://github.com/theia-ide/theia/pull/1886/files).

Reply:
It is fine since an object an object has a lifetime of an application.